bpf/lib/nodeport.h

```diff
@@ -1469,15 +1470,17 @@ static __always_inline int nodeport_snat_fwd_ipv4(struct __ctx_buff *ctx,
 	snat_needed = snat_v4_prepare_state(ctx, &target);
 	if (snat_needed)
 		ret = snat_v4_nat(ctx, &target, trace, ext_err);
-	if (ret == NAT_PUNT_TO_STACK)
-		ret = CTX_ACT_OK;
 
 	/* If multiple netdevs process an outgoing packet, then this packets will
 	 * be handled multiple times by the "to-netdev" section. This can lead
 	 * to multiple SNATs. To prevent from that, set the SNAT done flag.
 	 */
-	ctx_snat_done_set(ctx);
+	if (!IS_ERR(ret))
+    		ctx_snat_done_set(ctx);
```

Comment:
Ah, I didn't realize this would get more tricky on older releases ...

I think we need the following. Otherwise we would flag the packet as SNAT-done when didn't even try to SNAT it.

```suggestion
	if (snat_needed) {
		ret = snat_v4_nat(ctx, &target, trace, ext_err);
		if (!IS_ERR(ret))
    			ctx_snat_done_set(ctx);
    	}
```

But feel free to drop this PR, I can pick up the backport and give it a closer look.

[just checked, even v1.14 still used this old construct :grimacing:]

Reply:
Well since it's already here and likely doesn't need too much work we can bring it in as part of these backports. Could you take a look at what's the needed change for v1.12, v1.13 and v1.14 and send me the diff?